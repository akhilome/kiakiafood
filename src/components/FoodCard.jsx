import React from 'react';
import PropTypes from 'prop-types';

const FoodCard = ({
  foodImage, foodName, id, foodPrice, isAdmin,
}) => (
  <div className="food-card">
    <img className="food-image" src={foodImage} alt={foodName} />
    <div className="food-details">
      <div data-foodid={id} className="food-details__legend">
        <h4>{foodName}</h4>
        <p>
â‚¦
          {foodPrice.toLocaleString()}
        </p>
      </div>
      <div className="food-details__action">
        <button type="button">{isAdmin ? 'delete' : 'buy'}</button>
      </div>
    </div>
  </div>
);

FoodCard.propTypes = {
  foodImage: PropTypes.string.isRequired,
  foodName: PropTypes.string.isRequired,
  foodPrice: PropTypes.number.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  id: PropTypes.number.isRequired,
};

export default FoodCard;
