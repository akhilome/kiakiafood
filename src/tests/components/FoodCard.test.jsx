import React from 'react';
import { shallow } from 'enzyme';

import FoodCard from '../../components/FoodCard';

describe('Food Card Component', () => {
  const props = {
    foodImage: 'some.img',
    foodName: 'final fantasy',
    id: 69,
    foodPrice: 1996,
    isAdmin: false,
    buttonCallback: jest.fn(),
  };

  it('should render correctly for customer', () => {
    const wrapper = shallow(<FoodCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly for admin', () => {
    const wrapper = shallow(<FoodCard {...{ ...props, isAdmin: true }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
