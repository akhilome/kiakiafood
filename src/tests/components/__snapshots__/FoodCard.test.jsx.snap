// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Card Component should render correctly for admin 1`] = `
<div
  className="food-card"
>
  <img
    alt="final fantasy"
    className="food-image"
    src="some.img"
  />
  <div
    className="food-details"
  >
    <div
      className="food-details__legend"
      data-foodid={69}
    >
      <h4>
        final fantasy
      </h4>
      <p>
        ₦
        1,996
      </p>
    </div>
    <div
      className="food-details__action"
    >
      <button
        type="button"
      >
        delete
      </button>
    </div>
  </div>
</div>
`;

exports[`Food Card Component should render correctly for customer 1`] = `
<div
  className="food-card"
>
  <img
    alt="final fantasy"
    className="food-image"
    src="some.img"
  />
  <div
    className="food-details"
  >
    <div
      className="food-details__legend"
      data-foodid={69}
    >
      <h4>
        final fantasy
      </h4>
      <p>
        ₦
        1,996
      </p>
    </div>
    <div
      className="food-details__action"
    >
      <button
        type="button"
      >
        buy
      </button>
    </div>
  </div>
</div>
`;
